tc/sgx/trusted_worker_manager/enclave/work_order_processor_wpe.cpp

```diff
@@ -0,0 +1,289 @@
+            "verifyingKey", nullptr);
+
+        // resultUri is optional field. Hence don't throw exception
+        // if param is not there or empty in the request.
+        result_uri = GetJsonStr(params_object,
+            "resultUri", nullptr);
+
+        // notifyUri is optional field. Hence don't throw exception
+        // if param is not there or empty in the request.
+        notify_uri = GetJsonStr(params_object,
+            "notifyUri", nullptr);
+
+        work_order_id = GetJsonStr(params_object,
+            "workOrderId", "invalid request; failed to retrieve work order id");
+
+        worker_id = GetJsonStr(params_object,
+            "workerId", "invalid request; failed to retrieve worker id");
+
+        workload_id = GetJsonStr(params_object,
+            "workloadId", "invalid request; failed to retrieve work load id");
+
+        requester_id = GetJsonStr(params_object,
+            "requesterId", "invalid request; failed to retrieve requester id");
+
+        // workerEncryptionKey is optional field. Hence don't throw exception
+        // if param is not there or empty in the request.
+        worker_encryption_key = GetJsonStr(params_object,
+            "workerEncryptionKey", nullptr);
+
+        // dataEncryptionAlgorithm is optional field. Hence don't throw exception
+        // if param is not there or empty in the request.
+        data_encryption_algorithm = GetJsonStr(params_object,
+            "dataEncryptionAlgorithm", nullptr);
+
+        encrypted_session_key = GetJsonStr(params_object,
+            "encryptedSessionKey",
+            "invalid request; failed to retrieve encrypted session key");
+
+        session_key_iv = GetJsonStr(params_object,
+            "sessionKeyIv",
+            "invalid request; failed to retrieve session key iv");
+
+        requester_nonce = GetJsonStr(params_object,
+            "requesterNonce",
+            "invalid request; failed to retrieve requester nonce");
+
+        encrypted_request_hash = GetJsonStr(params_object,
+            "encryptedRequestHash",
+            "invalid request; failed to retrieve encrypted request hash");
+
+        // requesterSignature is optional field. Hence don't throw exception
+        // if param is not there or empty in the request.
+        requester_signature = GetJsonStr(params_object,
+            "requesterSignature", nullptr);
+
+        if (data_encryption_algorithm.length() > 0) {
+            tcf::error::ThrowIf<tcf::error::ValueError>(
+                data_encryption_algorithm != "AES-GCM-256",
+                "Unsupported dataEncryptionAlgorithm found in the request");
+        }
+
+        // Convert payload_format to lower case
+        std::transform(payload_format.begin(), payload_format.end(),
+            payload_format.begin(), ::tolower);
+        tcf::error::ThrowIf<tcf::error::ValueError>(
+            payload_format != "json-rpc",
+            "Unsupported payload format found in the input");
+
```

Comment:
Split the `ParseJsonInput()` in `work_order_processor.cpp`  into 1) `ParseJsonInput()`, 2) `decryptSessionKey()` and 3)`decryptDataKeys()` .
Use base class `ParseJsonInput()` in WPE and override the required function.

Reply:
Done